dcp/dry/class_manager.py

```diff
@@ -0,0 +1,46 @@
+import pythonmonkey as pm
+from .. import js
+from .. import api
+
+
+class ClassRegistry:
+    def __init__(self):
+        self._list = []
+
+    def _find(self, cmp):
+        return next((c for c in self._list if cmp(c)), None)
+
+    # TODO: this feels wrong, it's doing too many things. it should just add
+    # classes to the registry... instead it also does the inheritance stuff. ):
+    def add(self, bfclass):
+        if base_class := api.inheritance_hooks.get(bfclass.__name__):
```

Comment:
These aren't hooks. This is a map of BFClass wrapper names to subclasses. Maybe something like `api.sub_classes.get_by_name(...)`

Reply:
Idiot. Incompetent. Never submit an

Reply:
Severn you are so right, I also like the double entendre since not only are they "sub" calsses, but also "substitute" classes... :astonished: Wow! The definition of DRY can extend from code to language, why speak clearly when you can speak generically :money_mouth_face: double entendre